pkg/frontend/transport/handler.go

```diff
@@ -310,6 +320,15 @@ func formatQueryString(queryString url.Values) (fields []interface{}) {
 	return fields
 }
 
+func formatRequestHeaders(h *http.Header, headersToLog string) (fields []interface{}) {
```

Comment:
Could you add a test for this logic?

Reply:
https://github.com/grafana/mimir/pull/5030/commits/a280bc57cdc67f505e5d0d75bab24e499259d9fe